Azure/communication-ui-library-android
# [Refactor] make callbacks non inner classes

## Purpose

* Remove WeakReference on the Config (being disposed early), Updated Integration to migrate away from Anonymous Inner Classes (e.g. Error Handler and Token Function).

Weak Reference is in the Error Handler now, if the Launcher Activity dies while the composite is active, it'll still receive errors, but won't be able to notify the now dead activity. So worst case is reduced from a crash to not being able to notify that activity.

My leak canary is now super quiet as well (no notifications at all).


## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Run the application, join some calls. See if it works.
* Note: I haven't tested the token function totally, I'm not sure 100% how

```
```

## What to Check
Verify that the following are valid
* ...

## Other Information


The ViewModel still has a refresh token function, and I think probably could be refactored further to simplify, I believe it pulls a token in before starting a call.

I noticed the getToken in the ViewModel that was passed to the config before was just pulling Token from live data and not actually refreshing, but the Calling SDK said it's run in a handler (so I think it should be a sync network call that pulls new token instead of a cache). I'm not sure the impact of this totally.
